packages/simple-validators/src/validators.ts

```diff
@@ -207,22 +206,132 @@ export function validateEnum<T>(
   return input;
 }
 
+// This pattern implements the date pattern from ISO8601
+// Technically, it's also RFC3339 (a particular profile of ISO8601 i.e. YYYY-MM-DD
+// There is also a trailing capture group for timestamps
+const ISO8601_DATE_PATTERN = /^(\d\d\d\d)-(\d\d)-(\d\d)(T.*)?$/;
+// This pattern implements the following ABNF from RFC2822: `[ day-of-week "," ] date`
+// with a trailing capture group for time-like information
+const RFC2822_DATE_PATTERN =
+  /^(?:(Mon|Tue|Wed|Thu|Fri|Sat|Sun),)?\s*(\d{1,2})\s+(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s+(\d\d\d\d)\s*([^\s].*)?$/;
+
+const MONTH_TO_NUMBER = new Map(
+  ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'].map(
+    (elem, index) => [elem, index + 1],
+  ),
+);
+
+/**
+ * Build an ISO8601-compliant date string
+ */
+function buildISO8601DateString(year: number, month: number, day: number): string {
+  const paddedMonth = `${month}`.padStart(2, '0');
+  const paddedDay = `${day}`.padStart(2, '0');
+  return `${year}-${paddedMonth}-${paddedDay}`;
+}
+
 /**
  * Validate date string or object
  *
- * Uses javascript Date() constructor; any input to the constructor that returns
- * a valid date is valid input. This includes ISO 8601 formatted strings and
- * IETF timestamps are valid.
+ * Parses strings as ISO 8601 dates, or a variant of RFC 2822 dates, falling back to the Date constructor otherwise.
+ * Parses Date objects as UTC or local dates according to the given options.
+ * Parses `null` dates as "now".
```

Comment:
I think we are coming to the conclusion that this should not be allowed (at least from the frontmatter?)